apps/drained/app.ts

```diff
@@ -131,19 +131,15 @@ function drainedRestore() { // "public", to allow users to call
 }
 
 const checkCharge = () => {
-  if(E.getBattery() < restore) {
+  if(!Bangle.isCharging() || E.getBattery() < restore) {
     draw();
     return;
   }
   drainedRestore();
 };
 
-if (Bangle.isCharging())
-  checkCharge();
-
-Bangle.on("charging", charging => {
-  if(charging) checkCharge();
-});
+checkCharge();
+drainedInterval = setInterval(checkCharge, interval * 60 * 1000);
```

Comment:
Would it not be better to register and deregister the interval on the "charging" event?

Reply:
Great idea! Consider it done